constant_testing

```diff
@@ -110,17 +106,60 @@ clear_screen() {
   tput reset
 }
 
+elm_test_command() {
+  echo "elm-test"
+}
+
+elm_test() {
+  elm_json_path=$(find_first_elm_json)
+  elm_json_dir=$(dirname $elm_json_path)
+  pushd $elm_json_dir > /dev/null
+  run_command $(elm_test_command) "Your elm tests passed! First time for everything..." "Your elm is riddled with bugs... (at least you have tests(s) though)..." true
+  popd > /dev/null
+}
+
+run_command() {
+  cmd="$1"
+  success_message="$2"
+  failure_message="$3"
+  echo_output=$4
+  log_info "Running: $cmd"
+  output=`$cmd`
+  result=$?
+  last_run_cmd=$cmd
+  if [[ $echo_output = true ]]; then
+    echo "$output"
+  fi
+  if [[ $result != 0 ]]; then
+    log_failure "$failure_message"
+  else
+    log_success "$success_message"
+  fi
+}
+
+run_last_cmd() {
+  if [[ $last_run_cmd = $(elm_test_command) ]]; then
+    log_info "I don't know what to do with the file you just saved, so I'll just run the last thing I did again..."
+    elm_test
+  else
+    log_info "I don't know what to do with the file you just saved, so I'll just run the last thing I did again..."
+    run_command "$last_run_cmd" "Looks like everything is ok?" "It totally wrecked..." true
+  fi
+}
+
 watch() {
   clear_screen
-  if [ $FIXED_FILE ]; then
-    if [[ "$test_path" = *.exs ]]; then
+  if [ $ELM_TEST ]; then
```

Comment:
i think its definitely time to rewrite this conditional logic...

it should be something like 

is it a fixed file? 
dont care what file type etc. is  saved, run the file

is it an elixir file
 do the elixir stuff

is it an elm file
  do the elm stuff